### PR TITLE
common: Add Provider and set speaker action

### PR DESCRIPTION
**Pull request by thyal**
### Description

**This is a breaking change**
Adds a provider so that we can access the store in all components and hooks. This changes the API, and adds a requirement that the react hooks are rendered inside of a provider. That's why this is going to the development branch, and we will wait with releasing this until we are ready for `v3.0.0`. 

**Summary:**

**Related Issue:**

### Testing
1. Start storybook: `yarn build && yarn dev`
2. Go to one of the local media stories
3. Verify that the speaker selection works

### Checklist

-   [x] My code follows the project's coding standards.
-   [x] Prefixed the PR title and commit messages with the service or package name
-   [x] I have written unit tests (if applicable).
-   [x] I have updated the documentation (if applicable).
-   [x] By submitting this pull request, I confirm that my contribution is made
        under the terms of the MIT license.

### CODE REVIEW

File: packages/browser-sdk/src/lib/react/useRoomConnection/index.ts [hunk 1]
```diff
@@ -59,13 +58,12 @@ export function useRoomConnection(
     roomUrl: string,
     roomConnectionOptions = defaultRoomConnectionOptions,
 ): RoomConnectionRef {
-    const [store] = React.useState<Store>(() => {
-        if (roomConnectionOptions.localMedia) {
-            return roomConnectionOptions.localMedia.store;
```
Comment (richtrwhereby) [inline packages/browser-sdk/src/lib/react/useRoomConnection/index.ts:64]: Are we dropping this store override? Was it being used anywhere? 🤔

**Review by richtrwhereby — CHANGES_REQUESTED** [review]
Looking good and nice addition to the library 💯  

I made one code update suggestion and had a few other questions before we merge this. Please take a look and then I can re-review?

File: packages/browser-sdk/src/lib/react/useLocalMedia/index.ts [hunk 2]
```diff
@@ -1,11 +1,9 @@
-import { useCallback, useEffect, useState } from "react";
+import * as React from "react";
```
[the remark this answers is not in the document]
Comment (thyal) [reply packages/browser-sdk/src/lib/react/useLocalMedia/index.ts:1]: Not a strong reason. These destructed imports are totally fine, but the react team has said at some point they will stop supporting the default import of react (aka `import React from "react"`), so I default to this `import * as React`, and I like consistency... that's all. Happy to change it back if you want.

Side-note, there's a nice short article about this here: https://epicreact.dev/importing-react-through-the-ages/

File: packages/core/src/api/index.ts [hunk 3]
```diff
@@ -5,3 +5,4 @@ export { default as OrganizationApiClient } from "./OrganizationApiClient";
 export { default as OrganizationService } from "./organizationService";
 export { default as OrganizationServiceCache } from "./organizationServiceCache";
 export { default as RoomService } from "./roomService";
+export * from "./models/Organization";
```
[the remark this answers is not in the document]
Comment (thyal) [reply packages/core/src/api/index.ts:8]: Had some problems with the building of browser-sdk and importing of the `Store` type in the `Provider` component. This fixed it. But I can try to work around it.

File: packages/browser-sdk/src/lib/react/useRoomConnection/index.ts (hunk 1 above)
Comment (thyal) [reply packages/browser-sdk/src/lib/react/useRoomConnection/index.ts:64]: We don't need it any more. The reason this was here is that if you initialise local media hook first, and then the room connection hook, we need to use the same store. With the introduced Provider this is not an issue and we will have the same store anyways.

File: packages/core/src/api/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (thyal) [reply packages/core/src/api/index.ts:8]: This is the error I'm getting without these exports:
image

File: packages/browser-sdk/src/lib/react/useLocalMedia/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (richtrwhereby) [reply packages/browser-sdk/src/lib/react/useLocalMedia/index.ts:1]: No strong opinions at this point but I was curious why this was changed. Linked article was very interesting 👍

**Review by richtrwhereby — APPROVED** [review]

**Pull request merged**